### ISSUE TITLE
Live chart sometimes goes "saw tooth", "flips out" on update

### ISSUE THREAD

**Issue by strixy** [amcharts/amcharts4#3706]
**Question**

The first problem is that I don't even know what to call this behavior in order to search for more help. The best I can do is provide some screencaps.

In [the first chart](https://imgur.com/8df2fXt) you'll notice that everything is fine. [The second chart](https://imgur.com/8df2fXt) is that same chart only about 6 seconds later. Updates are pushed from the server via SSE every 2 - 10 seconds. The chart will sometimes go back to normal on the next update or (more commonly) after several updates. Usually the main chart returns to normal while the zoom bar stays borked. When that happens, the zoom bar will remain stuck like that until the chart is zoomed. Once zoomed, it doesn't seem to experience that problem anymore.

I am working with live data and initialize the chart with a decimated history to provide context. If I initialize the charts with less than 50 data points this doesn't happen. If I use 100 or more it always happens. (Back story, an entire historic dataset is anywhere between 200K to 2.0M points. We decimate it down to 50, 100 or whatever on the server side before sending it to the client. (Side note: I have actually load tested the charts using the full history up to about 500k and while it takes a little bit to load it does render it all which is pretty damn impressive so kudos on that achievement).

So, could this actually be performance related in some way? It does seem to go away the lower the number of data points I try to plot? I am not using the animation theme. I have also looked into the date parsing issue where performance is concerned and we're already sending in timestamps. With only 100 data points in such performant software I'm not sure what area of performance I should be looking at first? Could this be related to a large concentration of data points on the extreme right side of the chart? I wouldn't think so as it starts happening with only 1 live update, but I'll mention that thought just in case.

I'll keep trying to solve for performance issues. I found the section in the docs that covers that, thank you. I'll update here if I have any luck with that.

Am I adding the live data properly? eg. `series.addData()`

**Environment (if applicable)**

Tested Amcharts 4.10.20 and 4.10.21
Tested Chromium, Firefox, Edge (current and - 1)
Tested Linux and Windows 10
Running in Vue 2.

**Additional context**
This is the relevant code just in case I'm not doing something correctly.

```js
<template>
  <div class="amchart_wrapper" :ref="chartId">
  </div>
</template>

<script>
import * as am4core from '@amcharts/amcharts4/core'
import * as am4charts from '@amcharts/amcharts4/charts'
...
this.chart = am4core.create(this.$refs[this.chartId], am4charts.XYChart)
let dateAxis = this.chart.xAxes.push(new am4charts.DateAxis())
let scrollbarX = new am4charts.XYChartScrollbar()
...
this.cseries = this.chart.series.push(new am4charts.LineSeries())
this.cseries.data = cdata // a short version for the example
this.cseries.yAxis = this.valueAxis['cdata'] // created earlier elsewhere
scrollbarX.series.push(this.cseries)
--- 
// live data part
const ts = parseInt(d.timestamp)
this.cseries.addData({ date: ts, value: c }, 0)
...
</script>
```

**Comment by strixy** [comment]
I seem to have made improvements to the stability of the main chart when using `am4core.options.minPolylineStep = 15` (still testing various values) while also increasing the total initial data from 100 to 1000 which is also a great UX benefit as well. The zoom bar still goes bonkers, but this is an improvement.

**Comment by martynasma** [comment]
Would you be able to post your whole chart and data on CodePen/jsFiddle so that we can test?

**Comment by strixy** [comment]
Let me give it a shot and see what I can do.

**Comment by strixy** [comment]
Okay. Example created. If load the page and wait 3 seconds it'll update the chart and then the zoom bar (brush?) will go bonk. https://codesandbox.io/s/amcharts4-live-issue-hnlbh?file=/src/components/HelloWorld.vue

**Comment by martynasma** [comment]
Thanks. Seems like `addData` does not translate well to the scrollbar series clone.

We will look into if it can be fixed.

In the interim, you can fix it by adding same data point to the scrollbar series:

```JavaScript
this.cseries.addData({ date: ts, value: c }, 0);
this.chart.scrollbarX.scrollbarChart.series.getIndex(0).addData({ date: ts, value: c }, 0);
```

**Comment by strixy** [comment]
Thank you @martynasma! That seems to have it working nicely now. I appreciate all your help. I will look forward to the next release. :)

**Comment by martynasma** [comment]
Fixed in [4.10.22](https://github.com/amcharts/amcharts4/releases/tag/4.10.22).

## [4.10.22] - 2021-08-20

### Changed
- `package.json` for ES2015 package was properly updated with new version dependencies.

### Fixed
- Another issue with label text with a lot of in-line formatting not wrapping properly fixed.
- ForceDirectedTree was not updating label `maxWidth` on zoom.
- Legend's `position = "absolute"` was not working.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

[Download options](https://www.amcharts.com/download/).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
